src/Psalm/Config.php

```diff
@@ -709,14 +701,11 @@ public static function loadFromXMLFile($file_path, $current_dir): Config
 
     /**
      * Creates a new config object from an XML string
-     *
-     * @param  string           $base_dir
-     * @param  string           $file_contents
      * @param  string|null      $current_dir Current working directory, if different to $base_dir
      *
      * @throws ConfigException
      */
-    public static function loadFromXML($base_dir, $file_contents, $current_dir = null): Config
+    public static function loadFromXML(string $base_dir, string $file_contents, $current_dir = null): Config
```

Comment:
I think this needs to be promoted to `?string $current_dir = null` with corresponding docblock tag dropped. There are many instances of this syntax (old form of nullable params) so it might make sense to do them all at once.